x/stakeibc/keeper/msg_server_restore_interchain_account.go

```diff
@@ -0,0 +1,27 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Stride-Labs/stride/x/stakeibc/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k msgServer) RestoreInterchainAccount(goCtx context.Context, msg *types.MsgRestoreInterchainAccount) (*types.MsgRestoreInterchainAccountResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	hostZone, found := k.GetHostZone(ctx, msg.ChainId)
+	if !found {
+		k.Logger(ctx).Error(fmt.Sprintf("Host Zone not found: %s", msg.ChainId))
+		return nil, types.ErrInvalidHostZone
+	}
+
+	owner := types.FormatICAAccountOwner(msg.ChainId, msg.AccountType)
+	if err := k.ICAControllerKeeper.RegisterInterchainAccount(ctx, hostZone.ConnectionId, owner); err != nil {
```

Comment:
note: we're making an assumption that `ConnectionId` stored on the host zone matches what's in the IBC store. If we ever need to upgrade a connection (unsure if / why we'd have to do so), we'll need to also update the host zone.

I think we make this assumption in a lot of other places as well, just something for us to keep in mind